datafusion/sqllogictest/test_files/joins.slt

```diff
@@ -3269,22 +3262,17 @@ logical_plan
 10)----------TableScan: annotated_data projection=[a0, a, b, c, d]
 physical_plan
 01)SortPreservingMergeExec: [a@1 ASC, b@2 ASC NULLS LAST, c@3 ASC NULLS LAST, rn1@11 ASC NULLS LAST]
```

Comment:
And because there are 2 sorted partitions/streams after the join, we need this SortPreservingMergeExec to merge them